### ISSUE TITLE
Item merging: on sync pull create `item_link` record for `item`

### ISSUE THREAD

**Issue by Chris-Petty** [msupply-foundation/open-msupply#2553]
## Is your feature request related to a problem? Please describe 👀

During initialisation or regular synchronisation, items will be inserted into omSupply as we integrate records. But as we roll out `item_link` throughout the code base, we'll need `item_link` records to be generated for each item.

## Describe the solution you'd like 🎁

Base: [feature/merging](https://github.com/msupply-foundation/open-msupply/tree/feature/merging)

Side affect of during `insert` of an `item`, create an `item_link` with `item_link.id == item_link.item_id == item.id`

### Additional context 💌

In the distant future, the central server rewritten in rust will have item_link itself, and sync them out as central records.

### Moneyworks Jobcode 🧰

OMS:MERGE

**Comment by nisha-dangol** [comment]
Hi @Chris-Petty @roxy-dao .
Just confirming, is it expected for now to have both the items/names even after merging, unless we re-initialise the site? 
What I'm seeing is: on merging names(facilities) or items, I still see both the previous names/items. Only on re-initialising that site, I see the single merged name/item.
Video references if needed: [Reference1 for names](https://recordit.co/yGOlT8Ozyh), [Reference2 for items](https://recordit.co/JBpFs5pzM1)

**Comment by Chris-Petty** [comment]
@nisha-dangol Yup, old omSupply will still have both records. Reinitialising they come through merged, as on the central server the records are fully merged so when you get the fresh set of data you get the merged result.

I'm hoping in the next week we'll have a test build of omSupply's merging functionality at which point it should merge proper :)

**Comment by nisha-dangol** [comment]
> Oh I see your tag suggests that you did testing in dev? 
>
Not really. We use that tag `Build tested" Dev` when the test cases mentioned cannot be build-tested; assuming the devs have tested it while they approve/merge the changes! 
<br>
> If so you could try checking out branch `feature/merging` and testing with that, it should all work except for an edge case in shipment transfers at the moment (the fix is in PR). I will help make some comprehensive test suites next week though so no rush to test that branch right now.
>
Sure, I could give that a try! And I'm assuming the merging should work for all the items, patient names and facility names? Or, better I can test it after the PR get merged! 🤔

**Comment by Chris-Petty** [comment]
> > Oh I see your tag suggests that you did testing in dev?
> 
> Not really. We use that tag `Build tested: Dev` when the test cases mentioned cannot be build-tested; assuming the devs have tested it while they approve/merge the changes!
> 
> > If so you could try checking out branch `feature/merging` and testing with that, it should all work except for an edge case in shipment transfers at the moment (the fix is in PR). I will help make some comprehensive test suites next week though so no rush to test that branch right now.
> 
> Sure, I could give that a try! And I'm assuming the merging should work for all the items, patient names and facility names? Or, better I can test it after the PR get merged! 🤔

Yea really probably best to wait for `feature/merging` to get merged into `develop` as we're still churning a bit, so testing now might not be valid in a week :). I've been tryyying to remove the smaller incremental issues from the testing board - basically all issues with the "merging" tag shouldn't be tested, it'll be better to do a big test once it's settled. Which will be when the epic  #2450 is done.

**Comment by nisha-dangol** [comment]
> Yea really probably best to wait for `feature/merging` to get merged into `develop` as we're still churning a bit, so testing now might not be valid in a week :). I've been tryyying to remove the smaller incremental issues from the testing board - basically all issues with the "merging" tag shouldn't be tested, it'll be better to do a big test once it's settled. Which will be when the epic #2450 is done.
>
Cool, will wait until the Epic #2450 is merged!
